pkg/workceptor/remote_work.go

```diff
@@ -23,13 +23,14 @@ import (
 
 // remoteUnit implements the WorkUnit interface for the Receptor remote worker plugin.
 type remoteUnit struct {
-	BaseWorkUnit
+	// BaseWorkUnit
+	BaseWorkUnitForWorkUnit
 	topJC  *utils.JobContext
 	logger *logger.ReceptorLogger
 }
 
-// remoteExtraData is the content of the ExtraData JSON field for a remote work unit.
-type remoteExtraData struct {
+// RemoteExtraData is the content of the ExtraData JSON field for a remote work unit.
+type RemoteExtraData struct {
```

Comment:
I can see a lot of changes in remote_work.go here which is grand, but the gap in tests for some of these changes make me nervous.

A lot of the changes are fine, your accessing the logger in a new way. but some of the changes involve changing how we access the `status lock` and casting using the exported `RemoteExtraData` object. 
I would feel a lot more comfortable if these critical paths were tested in this PR.